predicators/envs/sandwich.py

```diff
@@ -0,0 +1,832 @@
+"""Sandwich making domain."""
+
+from typing import ClassVar, Dict, List, Optional, Sequence, Set, Tuple
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from gym.spaces import Box
+
+from predicators import utils
+from predicators.envs import BaseEnv
+from predicators.settings import CFG
+from predicators.structs import RGBA, Action, Array, GroundAtom, Object, \
+    ParameterizedOption, Predicate, State, Task, Type
+from predicators.utils import _Geom2D
+
+
+class SandwichEnv(BaseEnv):
+    """Sandwich making domain."""
+    # Parameters that aren't important enough to need to clog up settings.py
+    table_height: ClassVar[float] = 0.2
+    # The table x bounds are (1.1, 1.6), but the workspace is smaller.
+    x_lb: ClassVar[float] = 1.2
+    x_ub: ClassVar[float] = 1.5
+    # The table y bounds are (0.3, 1.2), but the workspace is smaller.
+    y_lb: ClassVar[float] = 0.4
+    y_ub: ClassVar[float] = 1.1
+    z_lb: ClassVar[float] = table_height - 0.02
+    z_ub: ClassVar[float] = z_lb + 0.25  # for rendering only
+    pick_z: ClassVar[float] = 0.7
+    robot_init_x: ClassVar[float] = (x_lb + x_ub) / 2
+    robot_init_y: ClassVar[float] = (y_lb + y_ub) / 2
+    robot_init_z: ClassVar[float] = pick_z
+    holder_width: ClassVar[float] = (x_ub - x_lb) * 0.8
+    holder_length: ClassVar[float] = (y_ub - y_lb) * 0.6
+    holder_x_lb: ClassVar[float] = x_lb + holder_width / 2
+    holder_x_ub: ClassVar[float] = x_ub - holder_width / 2
+    holder_y_lb: ClassVar[float] = y_lb + holder_length / 2
+    holder_y_ub: ClassVar[float] = holder_y_lb + (y_ub - y_lb) * 0.1
+    holder_color: ClassVar[RGBA] = (0.5, 0.5, 0.5, 1.0)
+    holder_thickness: ClassVar[float] = 0.01
+    board_width: ClassVar[float] = (x_ub - x_lb) * 0.8
+    board_length: ClassVar[float] = (y_ub - y_lb) * 0.2
+    board_x_lb: ClassVar[float] = x_lb + board_width / 2
+    board_x_ub: ClassVar[float] = x_ub - board_width / 2
+    board_y_lb: ClassVar[
+        float] = holder_y_lb + holder_length / 2 + board_length
+    board_y_ub: ClassVar[float] = board_y_lb + (y_ub - y_lb) * 0.05
+    board_color: ClassVar[RGBA] = (0.1, 0.1, 0.5, 0.8)
+    board_thickness: ClassVar[float] = 0.01
+    ingredient_thickness: ClassVar[float] = 0.02
+    ingredient_colors: ClassVar[Dict[str, Tuple[float, float, float]]] = {
+        "bread": (0.58, 0.29, 0.0),
+        "burger": (0.32, 0.15, 0.0),
```

Comment:
minor thing, but could we rename this to 'patty'? I was confused for a bit thinking that this object referred to the entire assembled thing (i.e, the bread + the patty + the veggies...)